pandas/core/reshape/pivot.py

```diff
@@ -148,8 +148,14 @@ def pivot_table(data, values=None, index=None, columns=None, aggfunc='mean',
 
     table = agged
     if table.index.nlevels > 1:
-        to_unstack = [agged.index.names[i] or i
-                      for i in range(len(index), len(keys))]
+        index_names = agged.index.names[:len(index)]
```

Comment:
add a comment here about level names being in integer could be ambiguous (and ref the issue number about this; I think we have one)